internal/document/frontmatter.go

```diff
@@ -1,19 +1,29 @@
 package document
 
 import (
+	byteslib "bytes"
 	"encoding/json"
 	"errors"
+	"fmt"
 	"strings"
 
 	"github.com/pelletier/go-toml/v2"
 	parserv1 "github.com/stateful/runme/internal/gen/proto/go/runme/parser/v1"
+	"github.com/stateful/runme/internal/idgen"
+	"github.com/stateful/runme/internal/version"
 	"gopkg.in/yaml.v3"
 )
 
+type RunmeMetaData struct {
```

Comment:
It's a bit odd that `Frontmatter` has only `yaml` tags, but this struct has more. Is there a reason for that?

Reply:
Likely because `yaml` is the most common frontmatter encoding we encounter. I've been meaning to test json and toml.

Reply:
In fact, @joseeantonior, could you please track a ticket for us to test `json` & `toml` frontmatter encoding, just so we don't forget? Thank you!

Reply:
https://github.com/stateful/runme/issues/415

Reply:
> There are tests that use TOML and JSON in their metadata, so I had to make sure they worked.

I specifically mean integration testing with the extension. Wasn't clear; will add that bit to the ticket.

Reply:
Nit: Ok, makes sense. Still, `protobuf` likely should be removed to avoid confusion; it is used only in generated stubs.